src/ipfsd-daemon.js

```diff
@@ -239,7 +239,24 @@ class Daemon {
       return this
     }
 
-    await this.api.stop()
+    let killTimeout
+    let killed = false
+    if (this.opts.forceKill !== false) {
+      killTimeout = setTimeout(() => {
+        // eslint-disable-next-line no-console
+        console.error(new Error(`Timeout stopping ${this.opts.type} node. Process ${this.subprocess.pid} will be force killed now.`))
+        killed = true
+        this.subprocess.kill('SIGKILL')
```

Comment:
can we use the this behaviour https://github.com/sindresorhus/execa#killsignal-options here ? send SIGTERM first and SIGKILL only if it doesnt die.